_docs/geyser/extensions.md

```diff
@@ -0,0 +1,123 @@
+```java
+@Subscribe
+public void onPostInitialize(GeyserPostInitializeEvent event) {
+    // example: show that your extension is loading.
+    this.logger().info("Loading example extension...");
+}
+```
+If you wish to register custom items, global resource packs (or soon, custom blocks and entities), you would need to subscribe to the event using the @Subscribe annotation,
+and register them in the event. You can find an example for custom items [here](/geyser/custom-items/#geyser-extensions). For other events, see [here](/geyser/events) for documentation.
+
+To build your extension, run the Gradle build task, and install the extension.
+
+### Creating commands with Geyser Extensions
+To create a command, you would need to use the "Commands" package in the Geyser API. Brief rundown:
+- [Command.java](https://github.com/GeyserMC/Geyser/blob/master/api/src/main/java/org/geysermc/geyser/api/command/Command.java)
+  This interface represents a command in Geyser - to make one, you can use the CommandBuilder. You can register it with the
+  [GeyserDefineCommandsEvent](https://github.com/GeyserMC/Geyser/blob/master/api/src/main/java/org/geysermc/geyser/api/event/lifecycle/GeyserDefineCommandsEvent.java)
+- [CommandExecutor.java](https://github.com/GeyserMC/Geyser/blob/master/api/src/main/java/org/geysermc/geyser/api/command/CommandExecutor.java)
+  This interface represents a command execute handler in Geyser, and extends the CommandSource interface.
+- [CommandSource.java](https://github.com/GeyserMC/Geyser/blob/master/api/src/main/java/org/geysermc/geyser/api/command/CommandSource.java)
+  This interface represents a command source in Geyser. It can be used to e.g. send messages to the source, check if the source has permission to execute a command, or get the locale.
+
+```java
+Command command = Command.builder(this) // "this" is the extension's main class
+                .name("ExampleCommand")
+                .bedrockOnly(true) // only allow bedrock players to execute this command
+                .source(CommandSource.class) // set the source of the command
+                .aliases(List.of("example", "ex")) // set aliases for the command
+                .description("An example command") // set the description of the command
+                .executableOnConsole(false) 
+                .suggestedOpOnly(false)
+                .permission("example.command") // set the permission for the command
+                .executor((source, cmd, args) -> {
+                    // this is the command executor - this is where you would put your code to execute the command.
+                    // source is the source that executed the command
+                    // cmd is the command that was executed
+                    // args are the arguments passed to the command
+                    source.sendMessage("Hello World");
+                })
+                .build();
```

Comment:
```suggestion
Command command = Command.builder(this) // "this" is the extension's main class
        .name("ExampleCommand")
        .bedrockOnly(true) // only allow bedrock players to execute this command
        .source(CommandSource.class) // set the source of the command
        .aliases(List.of("example", "ex")) // set aliases for the command
        .description("An example command") // set the description of the command
        .executableOnConsole(false) 
        .suggestedOpOnly(false)
        .permission("example.command") // set the permission for the command
        .executor((source, cmd, args) -> {
            // this is the command executor - this is where you would put your code to execute the command.
            // source is the source that executed the command
            // cmd is the command that was executed
            // args are the arguments passed to the command
            source.sendMessage("Hello World");
        })
        .build();
```
I also don't think that comments are really needed here. It's more to give a general feeling of how commands should be made and how you register them. For an explanation of the individual methods you have the javadocs.

Reply:
Reduced the amount of comments; i personally would leave in a few select ones about "this" and the executor